platform/android/java/lib/src/org/godotengine/godot/render/GLSurfaceView.java

```diff
@@ -1526,103 +1353,68 @@ private void guardedRun() throws InterruptedException {
 						continue;
 					}
 
-					if (createEglSurface) {
-						if (LOG_SURFACE) {
-							Log.w("GLThread", "egl createSurface");
-						}
-						if (mEglHelper.createSurface()) {
-							synchronized(sGLThreadManager) {
-								mFinishedCreatingEglSurface = true;
-								sGLThreadManager.notifyAll();
+					if (mRegisteredGLSurface != null) {
+						if (mRegisteredGLSurface.mFrameParams.createEglSurface) {
+							if (LOG_SURFACE) {
+								Log.w("GLThread", "egl createSurface");
 							}
-						} else {
-							synchronized(sGLThreadManager) {
-								mFinishedCreatingEglSurface = true;
-								mSurfaceIsBad = true;
-								sGLThreadManager.notifyAll();
+							if (mRegisteredGLSurface.mEglHelper.createSurface()) {
+								synchronized (sGLThreadManager) {
+									mRegisteredGLSurface.mFinishedCreatingEglSurface = true;
+									sGLThreadManager.notifyAll();
+								}
+							} else {
+								synchronized (sGLThreadManager) {
+									mRegisteredGLSurface.mFinishedCreatingEglSurface = true;
+									mRegisteredGLSurface.mSurfaceIsBad = true;
+									sGLThreadManager.notifyAll();
+								}
+								continue;
 							}
-							continue;
+							mRegisteredGLSurface.mFrameParams.createEglSurface = false;
 						}
-						createEglSurface = false;
-					}
-
-					if (createGlInterface) {
-						gl = (GL10) mEglHelper.createGL();
 
-						createGlInterface = false;
-					}
+						if (mRegisteredGLSurface.mFrameParams.createGlInterface) {
+							mRegisteredGLSurface.mFrameParams.gl = (GL10) mRegisteredGLSurface.mEglHelper.createGL();
 
-					// -- GODOT start --
-					if (createEglContext) {
-						if (LOG_RENDERER) {
-							Log.w("GLThread", "onSurfaceCreated");
+							mRegisteredGLSurface.mFrameParams.createGlInterface = false;
 						}
-						GLSurfaceView view = mGLSurfaceViewWeakRef.get();
-						if (view != null) {
+
+						if (mRegisteredGLSurface.mFrameParams.createEglContext) {
+							if (LOG_RENDERER) {
+								Log.w("GLThread", "onSurfaceCreated");
+							}
 							try {
-								view.mRenderer.onSurfaceCreated(gl, mEglHelper.mEglConfig);
+								mRenderer.onRenderSurfaceCreated(null);
```

Comment:
Why does this always pass `null` as the surface?

Reply:
Historically, it's because the native Android OpenGL logic didn't make use of the Android surface given that most of the OpenGL setup was done in the Java layer.